builtin/providers/google/resource_bigquery_dataset.go

```diff
@@ -0,0 +1,87 @@
+package google
+
+import (
+	"google.golang.org/api/bigquery/v2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceBigQueryDataset() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBigQueryDatasetCreate,
+		Read:   resourceBigQueryDatasetRead,
+		Update: resourceBigQueryDatasetUpdate,
+		Delete: resourceBigQueryDatasetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
```

Comment:
I am missing two general classes of things:
 - the role access information, which should be included.  I'll do that
 - updating/patching an existing dataset.  Update doesn't need to be implemented at all, in terraform's context that's handled by delete/create when a ForceNow config val is set.  I will use patch to implement Update

Anything else you'd like from that page?  I see list, which isn't useful in a terraform context, and a bunch of computed fields that don't have any obvious use in a terraform context.